### ISSUE TITLE
warp sync

### ISSUE THREAD

**Issue by xlc** [paritytech/substrate#14109]
Any docs about it? Can't find them.

To my understanding, it will download finality proof and put the best block to latest block and download historical blocks.
Is it possible to not download historical blocks? I tried with `--blocks-pruning=128 --state-pruning=128` and that doesn't seem to have any impacts.

Basically I want to minimize the resource requirement of a parachain node and I don't want to download and store all the relaychain data.

**Comment by xlc** [comment]
Had a quick check of the code and yeah I don't think it is currently possible to not download historical block. Can we change that?

**Comment by ggwpez** [comment]
I abused `--in-peers 0 --out-peers 0` to prevent that by restarting after it had downloaded everything.

**Comment by bkchr** [comment]
I still need to write some issue. I would say that we currently should disable blocks pruning at all. We need some more features to have this working properly. Otherwise we may loose block history.

**Comment by xlc** [comment]
My goal is to setup lightweight bootnodes that doesn't require too many resources (mostly disk) so we can have multiple of them distributed in different hosts. We have other RPC nodes to archive all blocks and states.

**Comment by bkchr** [comment]
> We have other RPC nodes to archive all blocks and states.

Yes, but the problem is that if enough people are running with blocks pruning other people needing to find these archive nodes in the network.

**Comment by xlc** [comment]
> We need some more features to have this working properly.

Could you explain a bit? Because otherwise I don't think the right solution is don't allow people to run lightweight node.

**Comment by bkchr** [comment]
Something like this: https://github.com/paritytech/substrate/issues/13248
